lib/generators/plugin/migration_generator.rb

```diff
@@ -0,0 +1,64 @@
+module Plugin
+    class MigrationGenerator < Rails::Generators::Base
+      class_option :plugin_name, :required => true
+      class_option :plugin_source
+
+      attr_accessor :migration_pattern
+
+      hook_for :orm, :required => true do |instance, klass|
+        new_migration = instance.invoke klass, nil, nil, destination_root: instance.send(:migration_destination)
+        instance.migration_pattern =  File.basename(new_migration.first).sub(/^\d*_/, '').sub(/\.rb$/, '')
+      end
+
+      def initialize(*args)
+        super
+
+        self.destination_root = migration_destination
+      end
+
+      def copy_latest_migration
+        destination_file = File.join(
+          destination_root,
+          'db/migrate',
+          destination_file_name
+        )
+
+        unless options['pretend']
+          copy_file migration_file, destination_file
+          remove_file File.join('db/migrate', migration_file)
+        end
+        [destination_file]
+      end
+
+      private
+
+      def source_paths
+        [File.join(migration_destination, 'db/migrate')]
+      end
+
+      def migration_destination
+        File.join(destination_root, plugin_source)
+      end
+
+      def destination_file_name
+        File.basename(migration_file, File.extname(migration_file)) +
+          ".#{plugin_name}" +
+          File.extname(migration_file)
+      end
+
+      def migration_file
+        @migration_file ||= begin
+          full_name = Dir.glob(File.join(migration_destination + 'db/migrate/*.rb')).grep(/\d+_#{migration_pattern}.rb$/).first
+          File.basename(full_name)
+        end
+      end
+
+      def plugin_source
+        @plugin_source ||= options[:plugin_source] || "../#{plugin_name}/"
```

Comment:
I see most of usage from developer setup (who else generates migrations?).
In case the folder structure is not standard - you can always specify the path by using `--plugin_source=../where/my/plugin/lives` option from the command line.